pkg/cluster/delete.go

```diff
@@ -231,6 +231,16 @@ func (m *manager) deleteResources(ctx context.Context) error {
 			m.log.Printf("deleting %s", *resource.ID)
 			future, err := m.resources.DeleteByID(ctx, *resource.ID, apiVersion)
 			if err != nil {
+				detailedError, ok := err.(autorest.DetailedError)
+				if ok {
+					if strings.Contains(detailedError.Original.Error(), "CannotDeleteLoadBalancerWithPrivateLinkService") {
+						err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeCannotDeleteLoadBalancerByID,
+							"features.ResourcesClient#DeleteByID", detailedError.Original.Error())
+					} else if strings.Contains(detailedError.Original.Error(), "AuthorizationFailed") {
+						err = api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden,
+							"features.ResourcesClient#DeleteByID", detailedError.Original.Error())
+					}
+				}
```

Comment:
this probably should be in a function, that is a lot of ifs here

Reply:
Moved to its own function and changed to a switch statement